File: server/scripts/seed-admin.ts
import { getConfig } from '../src/config.ts';
import { hashPassword } from '../src/auth/password.ts';
import { createPool, ensureSchema } from '../src/db/pool.ts';
import * as q from '../src/db/queries.ts';
import { ROLES } from '../src/types.ts';

async function main(): Promise<void> {
  const username = process.argv[2] ?? 'admin';
  const password = process.argv[3] ?? 'changeme123';
  const config = getConfig();
  console.log(`Connecting to PostgreSQL: ${config.database.user}@${config.database.host}:${config.database.port}/${config.database.name}`);

  const pool = createPool(config.database);
  try {
    await ensureSchema(pool);
    const existing = await q.getUserByUsernameExact(pool, username);
    if (!existing) {
      await q.insertUser(pool, {
        name: 'Super Admin',
        username,
        passwordHash: hashPassword(password),
        role: ROLES.superadmin,
      });
      console.log(`Superadmin created: username=${username} password=${password}`);
    } else {
      await q.updateUser(pool, existing.id, {
        passwordHash: hashPassword(password),
        role: ROLES.superadmin,
      });
      console.log(`Superadmin "${username}" already existed — password reset to: ${password}`);
    }
  } finally {
    await pool.end();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: server/tool/seed_admin.dart
import 'dart:io';

import 'package:drift/drift.dart';
import 'package:server/server.dart';

/// Seeds (or resets) the superadmin account.
///
///   dart run tool/seed_admin.dart                 # admin / changeme123
///   dart run tool/seed_admin.dart myuser mypass   # custom credentials
///
/// Requires PostgreSQL to be running and database `aclc` to exist
/// (schema `ssc` is created automatically on connect).
///
/// If the username already exists its password is reset — handy if you've
/// locked yourself out.
Future<void> main(List<String> args) async {
  final username = args.isNotEmpty ? args[0] : 'admin';
  final password = args.length > 1 ? args[1] : 'changeme123';

  final config = AppConfig.instance;
  stdout.writeln('Connecting to PostgreSQL: ${config.database.display}');

  final db = AppDatabase(config: config.database);
  try {
    final existing = await (db.select(
      db.users,
    )..where((u) => u.username.equals(username))).getSingleOrNull();

    if (existing == null) {
      await db
          .into(db.users)
          .insert(
            UsersCompanion.insert(
              name: 'Super Admin',
              username: username,
              passwordHash: PasswordHasher.hash(password),
              role: Roles.superadmin,
            ),
          );
      stdout.writeln(
        'Superadmin created: username=$username password=$password',
      );
    } else {
      await (db.update(db.users)..where((u) => u.id.equals(existing.id))).write(
        UsersCompanion(
          passwordHash: Value(PasswordHasher.hash(password)),
          role: const Value(Roles.superadmin),
          updatedAt: Value(pgNow()),
        ),
      );
      stdout.writeln(
        'Superadmin "$username" already existed — password reset to: $password',
      );
    }
  } finally {
    await db.close();
  }
}
